src/UiPath.Workflow.Runtime/Expressions/CompiledExpressionInvoker.cs

```diff
@@ -192,8 +192,21 @@ private bool TryGetCurrentCompiledExpressionRoot(ActivityContext activityContext
     private bool CanExecuteExpression(ICompiledExpressionRoot compiledExpressionRoot, out int expressionId)
     {
         var resultType = _expressionActivity.ResultType;
+
+        string normalizedExpressionText = _textExpression.ExpressionText;
+
+        if (_textExpression.Language == "VB")
+        {
+            //These characters are not allowed in VB code, so when VB compilation is done, they are replaced with normal quotes.
+            //But in the Expression Tree, we can have them, since they can be parsed directly from xaml code, or from a C# dll,
+            // so we need to "normalize" them, so that the comparison will work. 
+            normalizedExpressionText = normalizedExpressionText
+                .Replace('“', '"')
+                .Replace('”', '"');
+        }
+
```

Comment:
I was looking at this and made me wonder: are we placing the fix in the right place? I mean, sure, it is the most elegant approach, that gives us the benefit of not needing to re-compile existing projects and would work out-of-the-box, but on the other hand, I feel like the proper approach is to normalize the text closer to the design-experience, more precisely - when adding the text via the textbox, or when setting a ModelProperty value (in case of viewmodels). @vuplea, @marius-bughiu what do you think?

Reply:
Ideally, we would not allow them to begin with (we'd throw a validation error), but that would be a breaking change.
We can look at normalizing the values design-time as well, but I think we still need this one as well in order to support existing projects (and projects which will be migrated).

Reply:
so if we all agree its the best compromise, let's merge it. Indeed it brings the above-mentioned benefits 👍